Python/ceval.c

```diff
@@ -1551,6 +1551,12 @@ eval_frame_handle_pending(PyThreadState *tstate)
 #define DECREMENT_ADAPTIVE_COUNTER(cache) \
     (cache)->counter -= (1<<ADAPTIVE_BACKOFF_BITS)
 
+#define ADAPTIVE_COUNTER_CODE_IS_ZERO(codeunit) \
+    *codeunit < (1<<ADAPTIVE_BACKOFF_BITS)
+
+#define DECREMENT_ADAPTIVE_COUNTER_CODE(codeunit) \
+    *codeunit -= (1<<ADAPTIVE_BACKOFF_BITS)
```

Comment:
IMO all these macros should use more parentheses -- around the overall macro body, as well as around the argument where it occurs in the body. Macros should *always* be coded defensively like that. (The two macros above too, unless @markshannon disagrees.)